source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -193,8 +193,11 @@ bool DnsMessageParser::parseDnsObject(DnsQueryContextPtr& context,
     }
```

Comment:
If as below, that we are simply ignoring parsing the Answers/authority/additional RRs in a DNS client request, should we still set these states? Or after parsing questions section, then directly return?

Reply:
> If as below, that we are simply ignoring parsing the Answers/authority/additional RRs in a DNS client request, should we still set these states? Or after parsing questions section, then directly return?

We should still parse them I believe, because this state machine is to parse header to get the meta data like the number of questions/answers/authority/additional RRs. We can check if the request is expected based the header data, and then data are parsed from buffer like a byte stream.

Reply:
SG